### ISSUE TITLE
requiring malabar-mode fails with (invalid-slot-type ede-project-autoload proj-root function ede-maven2-project-root)

### ISSUE THREAD

**Issue by dakrone** [m0smith/malabar-mode#136]
When using a nightly CEDET (I believe this issue has to do with EIEIO), I get this error when requiring malabar-mode in my Emacs initialization:

```
(invalid-slot-type ede-project-autoload proj-root function ede-maven2-project-root)
```

Here is the full debugging trace:

```
Debugger entered--Lisp error: (invalid-slot-type ede-project-autoload proj-root function ede-maven2-project-root)
  signal(invalid-slot-type (ede-project-autoload proj-root function ede-maven2-project-root))
  eieio-validate-slot-value(ede-project-autoload 8 ede-maven2-project-root proj-root)
  eieio-oset([object ede-project-autoload "malabar-maven2" "MALABAR MAVEN2" ede/maven2 "pom.xml" t nil unbound nil unbound unbound nil t t] proj-root ede-maven2-project-root)
  #[(obj slots) "\304\216\305H B\n\2057
```

**Comment by m0smith** [comment]
In working on gh-177 I came across the same problems.  I will now be able to implement your suggestions.  Thanks

**Comment by Ingli** [comment]
I ran into a similar issue and couldn't solve it. What I did on Ubuntu 14.04 was to
- install Emacs 24.5.1 and use the builtin CEDET
- install sdkman from sdkman.io using the instructions there
- sdk install gradle groovy maven
- symlink ~/.gvm to ~/.sdkman
- symlink ~/.sdkman/candidates/groovy to ~/.sdkman/groovy
- in emacs, install using ELPA: dash, groovy-mode, malabar-mode

The error is the same as stated above: cannot find ede/maven2. Is there any hope to make malabar and emacs friends again?
